test/Dialect/HLS/round-trip.mlir

```diff
@@ -0,0 +1,11 @@
+// RUN: circt-opt %s -verify-diagnostics | circt-opt -verify-diagnostics | FileCheck %s
+
+module {
+  func @foo(%arg0 : index, %arg1 : index) {
+    // CHECK:      %0 = addi %arg0, %arg1 {hls.dir = #hls.pipeline<{II = 2 : i64, style = 2 : i64}>} : index
+    // CHECK-NEXT: %1 = addi %arg0, %arg1 {hls.dir = #hls.function_instantiate<{variable = "abc"}>} : index
+    %0 = addi %arg0, %arg1 {hls.dir = #hls.pipeline<{II = 2, style = 2}>} : index
+    %1 = addi %arg0, %arg1 {hls.dir = #hls.function_instantiate<{variable = "abc"}>} : index
```

Comment:
Seems like a minimal set here... Can you sketch out some of the more complicated use cases?  (i.e. an SCF loop with pipeline attribute?  a function call in an SCF loop?  basic blocks that have been lowered from an SCF loop?)  I see this as pretty experimental at the moment and maybe deserves some more experimentation before it gets committed.

Reply:
I do have more directives locally, however, opted to keep the number of directives in this PR to a minimum to ensure that we nail down how we want to represent the directives.

If it helps with exploring our options for representation, then i can add more directives.